src/reporting/ReportingWindow.sol

```diff
@@ -262,7 +278,7 @@ contract ReportingWindow is DelegationTarget, Typed, Initializable, IReportingWi
     }
 
     function isDoneReporting(address _reporter) public afterInitialized constant returns (bool) {
-        return getReportsByReporter(_reporter).count() >= getTargetReportsPerReporter();
+        return reporterDoneReporting[_reporter] || (totalMarketsThatRequiredReporting == 0);
```

Comment:
For reasons*, we cannot ever have a reporting window with 0 markets being reported on, so I don't think this check is necessary.

* We need to be able to identify who showed up for reporting and who didn't so we can correctly reward people that showed up during the month, even if they didn't have to do any meaningful work beyond blindly clicking a button in the UI.

Reply:
Ok! For now should I just comment out tests that would fail since we allow this case? If we know roughly how we want to implement this I could also just do so.

Reply:
I don't think we have a plan on the exact details of how to do this.  The theory is that if there are no markets in a reporting window when the reporting window starts, a placeholder market will be reported on with a question like, "Report YES if you want to receive reporting fees this period."  This market likely won't be able to go through the normal market creation codepath since it'll be created _after_ the reporting window starts.  Maybe now that we are using interfaces we can actually have the reporting window create an `IMarket` with a different implementation?

Alternatively we could come up with some other solution for allowing the user to sign an "I'm here" transaction for the special case of no markets to be reported on.  This may be simpler, but it would require the UI to have a separate code path for "no markets available".

Easiest solution test wise is to just (for now) have all tests include at least one market to be reported on and have one test that is skipped that has no markets to report on (since the functionality isn't implemented yet).

If you have some ideas on how to implement the faux market it may be a good next task, though it may be too big for this PR (depending on what the final solution is).

Reply:
🤔 more on it, does everything "work" with this change alone with regards to no markets?  If so, maybe we should leave this in for now until we implement the better solution later and just add a FIXME?

Reply:
I think for now I'll remove that logic since it'll end up being a turd and comment out the test with a note.

IMO the cleanest solution would be that every reporting window generates a special IMarket on construction that can only be reported on in the case of no reportable markets. This way no transaction is required to create the special market and it _should_ just work passively by having the appropriate functions that retrieve markets handle this.

I agree that should probably go in another PR though if that sounds reasonable.

Reply:
Hmm, actually... reporting windows are only created when someone creates a market, so maybe all of this is moot since there will be no reporting window in the first place.  :)

Reply:
That market may not need reporting though so I think it's still a thing